pkg/checks/latency.go

```diff
@@ -57,11 +59,8 @@ func (l *Latency) Run(ctx context.Context) error {
 		case <-l.done:
 			return nil
 		case <-time.After(l.cfg.Interval):
-			results, err := l.check(ctx)
+			results := l.check(ctx)
 			errval := ""
-			if err != nil {
-				errval = err.Error()
-			}
 			checkResult := Result{
 				Data:      results,
 				Err:       errval,
```

Comment:
We could now remove this field as it is part of the `Data` field in both checks.

Reply:
Possibly, but not as part of this PR. Why is the error in the Data as well, if there's already an `Err` field on the general `Result` struct? Is it important to store extra information on the error in the `Data` itself? Why is the `Result.Err` a `string` and not an `error`? Let's solve those questions first and then we can definetely move on from there.

Reply:
Result.Err is a string because it is directly exposed by the Api.

Currently it is not used. We should consider using it when something in general is not working for the check (like in the registration part). We should not forget about this but should be part of [issue]().